oauthlib/openid/connect/core/grant_types/refresh_token.py

```diff
@@ -0,0 +1,36 @@
+"""
+oauthlib.openid.connect.core.grant_types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+import logging
+
+from oauthlib.oauth2.rfc6749.grant_types.refresh_token import (
+    RefreshTokenGrant as OAuth2RefreshTokenGrant,
+)
+
+from .base import GrantTypeBase
+
+log = logging.getLogger(__name__)
+
+
+class RefreshTokenGrant(GrantTypeBase):
+
+    def __init__(self, refresh_id_token=True, request_validator=None, **kwargs):
+        self.refresh_id_token = refresh_id_token
+        self.proxy_target = OAuth2RefreshTokenGrant(
+            request_validator=request_validator, **kwargs)
+        self.register_token_modifier(self.add_id_token)
+
+    def add_id_token(self, token, token_handler, request):
+        """
+        Construct an initial version of id_token, and let the
+        request_validator sign or encrypt it.
+
+        The authorization_code version of this method is used to
+        retrieve the nonce accordingly to the code storage.
+        """
+        # Treat it as normal OAuth 2 auth code request if openid is not present
+        if not self.refresh_id_token:
```

Comment:
I don't see how do we set this boolean ? in the unit test it is done explicitely, but actually it should be done accordingly to if the scope includes "openid", right ?

Reply:
It is set during the initialization of the class (https://github.com/oauthlib/oauthlib/pull/752/files#diff-a9b30143f6cf03270ce00f8311710d82eaad0072eb5c01dd3aef15412c70ff68R19) and defaults to `True` so that the user can omit refreshing id tokens if he wishes to

Reply:
I see a problem if the Token Endpoint is used for both OAuth2.0 refresh and OIDC refresh. When this grant receives a refresh request, it has no dynamic choice to returns an id_token or not.

Reply:
What kind of dynamic logic do you have in mind?

If this grant is used for both OAuth2 and OIDC and you want to issue ID Tokens on refresh then this param is enough. `add_id_token` adds an ID Token in the response only if the `openid` scope was requested.

In addition if the client removes the `openid` scope in this refresh request then an ID Token will not be issued, which I think is the correct behaviour.

Do you think I should add more tests to validate this?